pkg/register/register.go

```diff
@@ -37,13 +38,21 @@ import (
 )
 
 func Register(url string, caCert []byte, smbios bool, emulateTPM bool, emulatedSeed int64) ([]byte, error) {
-
 	tpmAuth := &tpm.AuthClient{}
 	if emulateTPM {
 		logrus.Info("Enable TPM emulation")
 		if emulatedSeed == -1 {
-			emulatedSeed = rand.Int63()
-			logrus.Debugf("TPM emulation set to -1, setting random seed: %d", emulatedSeed)
+			data, err := ghw.Product(ghw.WithDisableWarnings())
+			if err != nil {
+				emulatedSeed = rand.Int63()
+				logrus.Debugf("TPM emulation using random seed: %d", emulatedSeed)
+			} else {
+				uuid := strings.Replace(data.UUID, "-", "", -1)
+				var i big.Int
+				i.SetString(uuid, 16)
```

Comment:
I think we should check if SetString() fails: otherwise if by chance  we got a failure, we may end up with a seed of 0 that would fail.

Reply:
Good point! Im pretty sure its probably impossible for this to fail as uuids are all the same format and so but probably it will fail in a very weird strange way :D

Reply:
done!